ext/standard/crypt.c

```diff
@@ -138,6 +152,33 @@ PHPAPI zend_string *php_crypt(const char *password, const int pass_len, const ch
 				ZEND_SECURE_ZERO(output, PHP_MAX_SALT_LEN + 1);
 				return result;
 			}
+		} else if (salt[0] == '$' && (salt[1] == 'y' || salt[1] == '7') && salt[2] == '$') {
+			yescrypt_local_t local;
+			uint8_t buf[PREFIX_LEN + 1 + HASH_LEN + 1]; /* prefix, '$', hash, NUL */
+
+			if (yescrypt_init_local(&local)) {
```

Comment:
I assume this function returns an error code instead of a success boolean? In that case I find an explicit comparison much clearer:

```suggestion
			if (yescrypt_init_local(&local) < 0) {
```

Same for `yescrypt_free_local()`.

Reply:
Yes it's a status.
This follows the same coding pattern as yescrypt's implementation already has: https://github.com/openwall/yescrypt/blob/e5873f89015c9c7c4ae56e3a6d17cefd47fef239/yescrypt-common.c#L480-L485
I prefer to keep that consistent, it's also safer if the API for error returns ever change.